crates/rolldown/src/stages/generate_stage/compute_cross_chunk_links.rs

```diff
@@ -182,7 +182,10 @@ impl<'a> GenerateStage<'a> {
                 }
               }
             })
-            .filter(|rec| matches!(rec.kind, ImportKind::Import))
+            .filter(|rec| {
+              matches!(rec.kind, ImportKind::Import)
+                && !rec.meta.contains(ImportRecordMeta::IS_EXPORT_START)
```

Comment:
a `export * from 'external'` should not be considered as a `importFromExteralModule' record, since it leads generate a redundant in the final codegen output. It is generated here already, https://github.com/rolldown/rolldown/blob/722fc095707828e5fbf5f94583258af744281546/crates/rolldown/src/ecmascript/format/esm.rs#L37-L50